pulsar-broker/src/test/java/org/apache/pulsar/schema/compatibility/SchemaTypeCompatibilityCheckOnTopicLevelTest.java

```diff
@@ -0,0 +1,134 @@
+
+        super.internalSetup();
+
+        // Setup namespaces
+        admin.clusters().createCluster(CLUSTER_NAME, ClusterData.builder().serviceUrl(pulsar.getWebServiceAddress())
+                .build());
+
+        TenantInfo tenantInfo = TenantInfo.builder()
+                .allowedClusters(Collections.singleton(CLUSTER_NAME))
+                .build();
+        admin.tenants().createTenant(PUBLIC_TENANT, tenantInfo);
+        admin.namespaces().createNamespace(namespaceName, Sets.newHashSet(CLUSTER_NAME));
+
+    }
+
+    @AfterClass(alwaysRun = true)
+    @Override
+    public void cleanup() throws Exception {
+        super.internalCleanup();
+    }
+
+    private void assertSchemaAlwaysInCompatibleStrategyInTopicLevel(String topic)
+            throws PulsarServerException, PulsarAdminException, PulsarClientException {
+        String topicName = TopicName.get(
+                TopicDomain.persistent.value(),
+                PUBLIC_TENANT,
+                namespace,
+                topic
+        ).toString();
+
+        pulsar.getAdminClient().topics().createNonPartitionedTopic(topicName);
+        pulsar.getAdminClient().topicPolicies().setSchemaCompatibilityStrategy(topicName,
+                SchemaCompatibilityStrategy.ALWAYS_INCOMPATIBLE);
+
+        Awaitility.await()
+                .untilAsserted(
+                        () -> assertEquals(
+                                pulsar.getAdminClient().topicPolicies().getSchemaCompatibilityStrategy(topicName),
+                                SchemaCompatibilityStrategy.ALWAYS_INCOMPATIBLE));
+
+        pulsarClient.newProducer(Schema.AVRO(SchemaDefinition.<Schemas.PersonOne>builder().
+                        withAlwaysAllowNull(true).withPojo(Schemas.PersonOne.class).build()))
+                .topic(topicName)
+                .create();
+
+        ProducerBuilder<Schemas.PersonThree> producerBuilder = pulsarClient.newProducer(
+                        Schema.AVRO(SchemaDefinition.<Schemas.PersonThree>builder().withAlwaysAllowNull(true)
+                                .withPojo(Schemas.PersonThree.class).build()))
+                .topic(topicName);
+
+        Throwable t = expectThrows(PulsarClientException.IncompatibleSchemaException.class, producerBuilder::create);
+        assertTrue(t.getMessage().contains("org.apache.avro.SchemaValidationException: Unable to read schema"));
+    }
+
+    @Test
+    public void testSchemaAlwaysInCompatibleStrategyInTopicLevel()
+            throws PulsarClientException, PulsarServerException, PulsarAdminException {
+        conf.setSchemaCompatibilityStrategy(SchemaCompatibilityStrategy.UNDEFINED);
+        admin.namespaces().setSchemaCompatibilityStrategy(namespaceName, SchemaCompatibilityStrategy.UNDEFINED);
+        assertSchemaAlwaysInCompatibleStrategyInTopicLevel("testSchemaAlwaysInCompatibleStrategyInTopicLevel");
+    }
+
+    @Test
+    public void testSchemaAlwaysInCompatibleStrategyOverriderBrokerConfigInTopicLevel()
```

Comment:
Please add test case for different level policy changed online. Something like :
set broker level value and check...
set ns level value and check...
set topic level value and check...
unset topic level value and check...
unset ns level value and check...
unset broker level value and check...